src/components/Input/Input.tsx

```diff
@@ -37,55 +37,122 @@ class Input extends UIComponent<any, any> {
     /** Additional classes. */
     className: PropTypes.string,
 
+    /** A property that will change the icon on the input and clear the input on click on Cancel */
+    clearable: PropTypes.bool,
+
     /** Optional Icon to display inside the Input. */
     icon: customPropTypes.itemShorthand,
 
     /** Shorthand for creating the HTML Input. */
     input: customPropTypes.itemShorthand,
 
+    /**
+     * Called on change.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props and proposed value.
+     */
+    onChange: PropTypes.func,
+
     /** The HTML input type. */
     type: PropTypes.string,
+
+    /** The value of the input. */
+    value: PropTypes.string,
   }
 
-  static handledProps = ['as', 'children', 'className', 'icon', 'input', 'type']
+  static handledProps = [
+    'as',
+    'children',
+    'className',
+    'clearable',
+    'icon',
+    'input',
+    'onChange',
+    'type',
+    'value',
+  ]
 
   static defaultProps = {
     as: 'div',
     type: 'text',
   }
 
+  constructor(props, context) {
+    super(props, context)
+
+    this.state = {
+      value: props.value || '',
+    }
+
+    this.handleChange = this.handleChange.bind(this)
```

Comment:
Looks like useless line because `handleChange` is an arrow function.

https://medium.com/@machnicki/handle-events-in-react-with-arrow-functions-ede88184bbb

Reply:
fixed